ui/src/main/java/edu/wpi/grip/ui/pipeline/PipelineController.java

```diff
@@ -278,13 +273,13 @@ public void onStepMoved(StepMovedEvent event) {
     @Subscribe
     public void onConnectionAdded(ConnectionAddedEvent event) {
         // Add the new connection view
-        platform.runAsSoonAsPossible(() -> addConnectionView(event.getConnection()));
+        addConnectionView(event.getConnection());
     }
 
     @Subscribe
     public void onConnectionRemoved(ConnectionRemovedEvent event) {
         // Remove the control that corresponds with the connection that was removed
-        platform.runAsSoonAsPossible(() -> {
+        PlatformImpl.runAndWait(() -> {
```

Comment:
I don't have evidence that it doesn't work in this case no.
What I do know is that `runAndWait` can deadlock the javafx thread on the event bus because another thread is waiting on the javafx thread to run and the javafx thread is trying to run an event.

Basically, I don't trust `runAndWait` at all. I think it is a nasty hack around a problem that is a simple drawing problem. Try invalidating the view and triggering a redraw instead.